Public/Src/Tools/Xldb/README.md

```diff
@@ -0,0 +1,26 @@
+# XLDB DataStore
+
+The XLDB datastore contains the APIs that allow a user to access Xldb (the RocksDB instance that contains all of the log information).	
+It is readonly (will not add anything new to the DB).
+It will happily read in any Xldb version that is equal to or less than the current version. 
+It contains protobuf objects for keys and values so that these are also forward/backwards compatible!
+
+> The Xldb version is updated upon a breaking change in the underlying datastore or indexing (i.e. switching out of RocksDB or changing keys for a value).
+ 
+* A Nuget package (see below) contains this public facing api that a user can access to get data from the DB instance.
+	- A user can just import the Nuget package to their solution and use the APIs to crawl through the database
+	- Be careful of changing API behavior and/or properly deprecating old APIs once this is published so you don't cause breaking changes for users of this Nuget package
+
+All public facing API endpoints can be found under `IXldbDataStore.cs` with the neccessary comments and documentation.
+
+The datastore can be consumed as a Nuget package which contains all the neccesary dlls. This Nuget package is not tied to the engine version and it can be found here: https://dev.azure.com/1esSharedAssets/1esPkgs/_packaging?_a=feed&feed=BuildXL.

```

Comment:
I don't think we should advertise this since this feed is private

Reply:
Moving it to internal documentation

---
In reply to: [324344666](https://github.com/microsoft/BuildXL/pull/840#discussion_r324344666) [](ancestors = 324344666)